ReactAndroid/src/main/java/com/facebook/react/modules/fresco/FrescoModule.java

```diff
@@ -63,29 +62,39 @@ public FrescoModule(ReactApplicationContext reactContext) {
    */
   public FrescoModule(ReactApplicationContext reactContext, @Nullable ImagePipelineConfig config) {
     super(reactContext);
+    setConfig(config);
+  }
+
+  public void setConfig(@Nullable ImagePipelineConfig config) {
+    if (hasBeenInitialized()) {
+      FLog.w(ReactConstants.TAG,
+        "Fresco has already been initialized with a different config."
+        + "The new Fresco configuration will be ignored!");
+      return;
+    }
+
+    if (config == null) {
+      config = getDefaultConfig(this.getReactApplicationContext());
+    }
     mConfig = config;
   }
 
-  @Override
-  public void initialize() {
-    super.initialize();
+  public void initializeFresco() {
     if (!hasBeenInitialized()) {
       // Make sure the SoLoaderShim is configured to use our loader for native libraries.
       // This code can be removed if using Fresco from Maven rather than from source
       SoLoaderShim.setHandler(new FrescoHandler());
-      if (mConfig == null) {
-        mConfig = getDefaultConfig(getReactApplicationContext());
-      }
-      Context context = getReactApplicationContext().getApplicationContext();
-      Fresco.initialize(context, mConfig);
+      Fresco.initialize(this.getReactApplicationContext(), mConfig);
       sHasBeenInitialized = true;
-    } else if (mConfig != null) {
```

Comment:
I think this case can stay deleted with mConfig being set to null below. Looks like a refactor may have gone wrong.